### PR TITLE
allowing all plot functions within pulses.py to save the plot as a .png file

### PR DESCRIPTION
**Pull request by aorgazf**
This functionality is useful for debugging through the slurm queue

### CODE REVIEW

**Comment by DavidSarlle** [comment]
Looks goot to me. It works.

**Review by stavros11 — APPROVED** [review]
Indeed, this is useful, thanks.

**Comment by scarrazza** [comment]
Could you please try fix coverage so we can merge?

File: src/qibolab/pulses.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def __hash__(self):
     def __repr__(self):
         return self.serial
 
-    def plot(self):
+    def plot(self, savefig_filename=None):
```
Comment (scarrazza) [inline src/qibolab/pulses.py:37]: Could you please document this function?

File: src/qibolab/pulses.py [hunk 2]
```diff
@@ -838,7 +842,7 @@ def modulated_waveform_q(self) -> Waveform:
     def modulated_waveforms(self):  #  -> tuple[Waveform, Waveform]:
         return (self.modulated_waveform_i, self.modulated_waveform_q)
 
-    def plot(self):
+    def plot(self, savefig_filename=None):
```
Comment (scarrazza) [inline src/qibolab/pulses.py:845]: Docstrings required also here.

File: src/qibolab/pulses.py [hunk 3]
```diff
@@ -1155,7 +1161,7 @@ def pulses_overlap(self) -> bool:
                 overlap = True
         return overlap
 
-    def plot(self):
+    def plot(self, savefig_filename=None):
```
Comment (scarrazza) [inline src/qibolab/pulses.py:1164]: And here.

**Review by scarrazza — CHANGES_REQUESTED** [review]
Please include docstrings explaining what the method does and its arguments.

**Pull request merged**